client/pages/view-entries.jsx

```diff
@@ -106,7 +106,21 @@ function Entries(props) {
       <div className='caption'>
         <div className='view-entries-title-section'>
           <a href={`#code?entryId=${entryId}`} className='entries-anchor'><h2 className='view-entries-content-title'>{title}</h2></a>
+            {window.localStorage.getItem('userId') === userId.toString() &&
+          <div>
           <a href={`#edit-code?entryId=${entryId}`} className='entries-anchor'><i className="fas fa-edit adjust-editing-button"></i></a>
+            <a href="#entries" onClick={() => {
+              if (shared === 'no') {
+                fetch(`/api/share/${entryId}`, { method: 'PATCH', headers: { 'Content-Type': 'application/json', 'x-access-token': window.localStorage.getItem('react-context-jwt') } });
+                setShared('yes');
+              }
+              if (shared === 'yes') {
+                fetch(`/api/noshare/${entryId}`, { method: 'PATCH', headers: { 'Content-Type': 'application/json', 'x-access-token': window.localStorage.getItem('react-context-jwt') } });
+                setShared('no');
+              }
+            }}><i className={shared === 'yes' ? 'fas fa-share-square share-icon' : 'fas fa-share share-icon'}></i></a>
```

Comment:
Your event listeners are starting to get messy. Try to define a function before the JSX part of the component and pass it to `onClick`.